lte/cloud/go/services/policydb/obsidian/handlers/policy_handlers.go

```diff
@@ -417,19 +395,74 @@ func DeleteRule(c echo.Context) error {
 	if err != nil {
 		return obsidian.HttpError(err, http.StatusInternalServerError)
 	}
+
 	return c.NoContent(http.StatusNoContent)
 }
 
-func getNetworkIDAndBaseName(c echo.Context) (string, string, *echo.HTTPError) {
-	vals, err := obsidian.GetParamValues(c, "network_id", baseNameParam)
+// QoS profiles
+
+func getQoSProfiles(c echo.Context) error {
+	networkID, nerr := obsidian.GetNetworkId(c)
+	if nerr != nil {
+		return nerr
+	}
+
+	profiles, err := configurator.LoadAllEntitiesInNetwork(networkID, lte.PolicyQoSProfileEntityType, configurator.EntityLoadCriteria{LoadConfig: true})
 	if err != nil {
-		return "", "", err
+		return obsidian.HttpError(err, http.StatusInternalServerError)
 	}
-	return vals[0], vals[1], nil
+
+	ret := map[string]*models.PolicyQosProfile{}
+	for _, ent := range profiles {
+		ret[ent.Key] = ent.Config.(*models.PolicyQosProfile)
+	}
+
+	return c.JSON(http.StatusOK, ret)
+}
+
+func createQoSProfile(c echo.Context) error {
+	networkID, nerr := obsidian.GetNetworkId(c)
+	if nerr != nil {
+		return nerr
+	}
+	profile := &models.PolicyQosProfile{}
+	if err := c.Bind(profile); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+	if err := profile.ValidateModel(); err != nil {
+		return obsidian.HttpError(err, http.StatusBadRequest)
+	}
+
+	exists, err := configurator.DoesEntityExist(networkID, lte.PolicyQoSProfileEntityType, profile.ID)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	if exists {
+		return echo.ErrNotFound
```

Comment:
I don't think 404 is the right error to be returning here

Reply:
Good catch